src/index.js

```diff
@@ -0,0 +1,152 @@
+  });
+  const button = document.createElement('button');
+  button.setAttribute('type', 'button');
+  button.classList.add('list-button');
+  button.innerHTML = 'Clear all completed';
+  mainList.appendChild(button);
+}
+
+function buttonListener() {
+  const checkboxes = document.querySelectorAll('.list-box');
+  Array.from(checkboxes).forEach(box => {
+    box.addEventListener('change', (event) => {
+      if (event.target.checked) {
+        checked(event.target.id, taskArray);
+      } else {
+        unchecked(event.target.id, taskArray);
+      }
+    });
+  });
+}
+
+function itemListener() {
+  const addField = document.querySelector('.list-input');
+  addField.addEventListener('click', event => {
+    event.target.setAttribute('class', 'list-input-clicked');
+  });
+  addField.addEventListener('focusout', event => {
+    event.target.setAttribute('class', 'list-input');
+  });
+  addField.addEventListener('keydown', e => {
+    if (e.key === 'Enter') {
+      addTask(taskArray, addField.value);
+    }
+  });
+}
+
+function descriptionListener() {
+  const descriptions = document.querySelectorAll('.description');
+  Array.from(descriptions).forEach((desc) => {
+    desc.addEventListener('click', event => {
+      const checkIfOpened = event.target.parentNode.getAttribute('value');
+      if (checkIfOpened !== null) {
+        return;
+      }
+      const value = event.target.getAttribute('value');
+      const initialState = event.target.innerHTML;
+      event.target.innerHTML = `<input class="description" id="input${value}" value="${taskArray[value].description}"></input>`;
+      const inputField = document.getElementById(`input${value}`);
+      inputField.focus();
+      const imageChange = document.querySelector(`img[value="${value}"]`);
+      const initialImage = imageChange.src;
+      imageChange.addEventListener('click', e => {
+        removeItem(e, taskArray);
+        mainList.dispatchEvent(forceChange);
+      });
+      imageChange.src = TrashIcon;
+      inputField.addEventListener('focusout', (e) => {
+        e.target.outerHTML = initialState;
+        imageChange.src = initialImage;
+      });
+      inputField.addEventListener('keydown', (e) => {
+        if (e.key === 'Enter') {
+          edit(event, taskArray, value);
+        }
+      });
+    });
+  });
+}
+
+function removeAllListener() {
+  const removeAllBtn = document.querySelector('.list-button');
+  removeAllBtn.addEventListener('click', () => {
+    taskArray = removeAll(taskArray);
+    mainList.dispatchEvent(forceChange);
+  });
+}
+
+document.addEventListener('DOMContentLoaded', paintList());
+document.addEventListener('DOMContentLoaded', buttonListener);
+document.addEventListener('DOMContentLoaded', itemListener());
+document.addEventListener('DOMContentLoaded', descriptionListener());
+document.addEventListener('DOMContentLoaded', removeAllListener());
+mainList.addEventListener('change', paintList);
+mainList.addEventListener('change', () => {
```

Comment:
[Optional] All the events for the same DOM element could be grouped in the same block code, avoiding repeating the code to comply with DRY principles.